server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -470,6 +472,25 @@ public boolean isRemoteStoreNode() {
         return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
     }
 
+    /**
+     * Returns whether the node is a remote cluster state enabled node.
+     * @return true if the node contains remote cluster state and remote routing table node attributes, false otherwise
+     */
+    public boolean isRemoteStateNode() {
+        return isRemoteClusterStateEnabled() && isRemoteRoutingTableEnabled();
```

Comment:
Publication should be optional

Reply:
This is being used to decide publication only as of now. Would it be better if we rename this to something like `isRemotePublicationEnabled`